### PR TITLE
Waffling60 Revision C

### PR DESCRIPTION
**Pull request by 4pplet**
## Description

Adding new revision of waffling60. QMK code has very minor changes compared to rev_b, but PCB has additional layout options, so it needs another ID for VIA.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/4pplet/waffling60/rev_c/config.h [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#define QMK_ESC_INPUT C4 // usually ROW
+
+#define RGB_DI_PIN D3
+#define RGBLED_NUM 1
+#define BACKLIGHT_LEVELS 5
+
+#define RGBLIGHT_HUE_STEP 8
+#define RGBLIGHT_SAT_STEP 8
+#define RGBLIGHT_VAL_STEP 8
+#define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/4pplet/waffling60/rev_c/config.h:55]: Could you explicitly define which animations are enabled? 
```suggestion
#define RGBLIGHT_EFFECT_BREATHING
#define RGBLIGHT_EFFECT_RAINBOW_MOOD
#define RGBLIGHT_EFFECT_RAINBOW_SWIRL
#define RGBLIGHT_EFFECT_SNAKE
#define RGBLIGHT_EFFECT_KNIGHT
#define RGBLIGHT_EFFECT_CHRISTMAS
#define RGBLIGHT_EFFECT_STATIC_GRADIENT
#define RGBLIGHT_EFFECT_RGB_TEST
#define RGBLIGHT_EFFECT_ALTERNATING
#define RGBLIGHT_EFFECT_TWINKLE
```

File: keyboards/4pplet/waffling60/rev_c/readme.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# waffling60
+
+A 60% PCB for MX switches, one hot swap and one solder-pcb version with decent layout support.
+
+More info: https://geekhack.org/index.php?topic=103531.0
+
+* Keyboard Maintainer: [4pplet](https://github.com/4pplet)
+* Hardware Supported: [waffling60](https://github.com/4pplet/waffling60)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/4pplet/waffling60/rev_c/readme.md:10]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by 4pplet** [comment]
Thanks! Made adjustments according to the comments 👍

File: keyboards/4pplet/waffling60/rev_c/rev_c.c [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        if(res) {
+            if(led_state.caps_lock){
+                rgblight_enable();
+                rgblight_mode(1);
+                rgblight_sethsv_noeeprom(CAPS_LOCK_COLOR);
+            }
+            else
+                rgblight_disable_noeeprom();
+        }
+    }
```
Comment (drashna) [inline keyboards/4pplet/waffling60/rev_c/rev_c.c:31]:
```suggestion
    if (CAPS_LOCK_ENABLE && res) {
        if (led_state.caps_lock){
            rgblight_enable();
            rgblight_mode(1);
            rgblight_sethsv_noeeprom(CAPS_LOCK_COLOR);
        } else {
            rgblight_disable_noeeprom();
        }
    }
```

Comment (4pplet) [reply keyboards/4pplet/waffling60/rev_c/rev_c.c:31]: Thanks! Adding this 👍

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Pull request merged**